torchgeo/datasets/stacapidataset.py

```diff
@@ -0,0 +1,321 @@
+            <https://pystac-client.readthedocs.io/en/latest/api.html#pystac_client.ItemSearch>
+        """
+        self.root = root
+        self.api_endpoint = api_endpoint
+        self.bands = bands
+        self.is_image = is_image
+
+        # Create an R-tree to index the dataset
+        self.index = Index(interleaved=False, properties=Property(dimension=3))
+
+        catalog = Client.open(api_endpoint)
+
+        search = catalog.search(**query_parameters)
+
+        items = list(search.get_items())
+
+        if not items:
+            raise RuntimeError(
+                f"Your search criteria off {query_parameters} did not return any items"
+            )
+
+        epsg = items[0].properties["proj:epsg"]
+        crs_dict = {"init": "epsg:{}".format(epsg)}
+        src_crs = CRS.from_dict(crs_dict)
+        if crs is None:
+            crs = CRS.from_dict(crs_dict)
+
+        for i, item in enumerate(items):
+            minx, miny, maxx, maxy = item.bbox
+
+            transformer = Transformer.from_crs(src_crs.to_epsg(), crs.to_epsg())
+            (minx, maxx), (miny, maxy) = transformer.transform(
+                [minx, maxx], [miny, maxy]
+            )
+            mint = 0
+            maxt = sys.maxsize
+            coords = (minx, maxx, miny, maxy, mint, maxt)
+            self.index.insert(i, coords, item)
+
+        self._crs = crs
+        self.res = 10
+        self.transforms = transforms
+        self.items = items
+
+    def __getitem__(self, query: BoundingBox) -> Dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image/mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        hits = self.index.intersection(tuple(query), objects=True)
+        items = [hit.object for hit in hits]
+
+        if not items:
+            raise IndexError(
+                f"query: {query} not found in index with bounds: {self.bounds}"
+            )
+
+        bounds = (query.minx, query.miny, query.maxx, query.maxy)
+
+        raster_list = []
+        for item in items:
+            raster_list.append(self._snap_to_single_raster(item, bounds))
+
+        # merge single rasters
+        data = self._merge_rasters(raster_list)
+
+        # if only single time step then squeeze out time dimenstion
+        image = data.squeeze(0)
+
+        # suggested #
+        signed_items = [pc.sign(item).to_dict() for item in items]
+
+        stack = stackstac.stack(
+            signed_items,
+            assets=self.bands,
+            resolution=self.res,
+            epsg=self._crs.to_epsg(),
+        )
+
+        aoi = stack.loc[
+            ..., query.maxy : query.miny, query.minx : query.maxx  # type: ignore[misc]
+        ]
```

Comment:
> @calebrob6 I have been trying out your suggestion with different bounds and stacks, but here the `.loc[]` indexing always returns an array of shape [num_items, channels, 0, 0] so no height or width, and I can't seem to figure out what is causing that.

This doesn't work because the `stack` DataArray has coordinates in a UTM projection, but the `query` was using longitude/latitude coordinates. Need to use the same coordinate reference system in both for this to work. See my suggestion at L113 that should fix this.